From: Julien Cristau <jcristau@debian.org>
Subject: greeter: call openlog() not just on openbsd

If we don't call openlog() pam will use syslog but won't have a service name.
See Debian bug #382037.

Note: this patch is not appropriate for upstream; the bug log contains a more
complete patch, but this one should be ok for debian's purposes.

Index: xdm/greeter/greet.c
===================================================================
--- xdm.orig/greeter/greet.c
+++ xdm/greeter/greet.c
@@ -83,9 +83,7 @@
 #include "greet.h"
 #include "Login.h"
 
-#ifdef __OpenBSD__
 #include <syslog.h>
-#endif
 
 #if defined(SECURE_RPC) && defined(sun)
 /* Go figure, there's no getdomainname() prototype available */
@@ -408,13 +406,9 @@
 static void
 FailedLogin (struct display *d, struct greet_info *greet)
 {
-#ifdef __OpenBSD__
-    syslog(LOG_NOTICE, "LOGIN FAILURE ON %s",
-	   d->name);
     syslog(LOG_AUTHPRIV|LOG_NOTICE,
 	   "LOGIN FAILURE ON %s, %s",
 	   d->name, greet->name);
-#endif
     DrawFail (login);
 #ifndef USE_PAM
     bzero (greet->name, strlen(greet->name));
@@ -485,9 +479,7 @@
 	LogError ("Cannot reopen display %s for greet window\n", d->name);
 	exit (RESERVER_DISPLAY);
     }
-#ifdef __OpenBSD__
-    openlog("xdm", LOG_ODELAY, LOG_AUTH);
-#endif
+    openlog("xdm", LOG_ODELAY | LOG_PID, LOG_AUTHPRIV);
 
     for (;;) {
 #ifdef USE_PAM
